### PR TITLE
Migrate away from deprecated text themes

### PR DESCRIPTION
**Pull request by kenzieschmoll**
The 2018 TextTheme properties have been deprecated in favor of the new 2021 properties.

### CODE REVIEW

File: packages/devtools_app/lib/src/framework/initializer.dart [hunk 1]
```diff
@@ -167,7 +167,7 @@ class _InitializerState extends State<Initializer>
           child: Column(
             children: [
               const Spacer(),
-              Text('Disconnected', style: theme.textTheme.headline3),
```
Comment (CoderDake) [inline packages/devtools_app/lib/src/framework/initializer.dart:170]: should this change to `headlineMedium`

File: packages/devtools_app/lib/src/framework/scaffold.dart [hunk 2]
```diff
@@ -452,7 +452,7 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
     final painter = TextPainter(
       text: TextSpan(
         text: title,
-        style: textTheme.headline6,
+        style: textTheme.titleLarge,
```
Comment (CoderDake) [inline packages/devtools_app/lib/src/framework/scaffold.dart:455]: Should this be `headlineSmall`?

File: packages/devtools_app/lib/src/shared/common_widgets.dart [hunk 3]
```diff
@@ -1222,7 +1222,7 @@ class CenteredMessage extends StatelessWidget {
     return Center(
       child: Text(
         message,
-        style: Theme.of(context).textTheme.headline6,
+        style: Theme.of(context).textTheme.titleLarge,
```
Comment (CoderDake) [inline packages/devtools_app/lib/src/shared/common_widgets.dart:1225]: should this be `headlineSmall`?

File: packages/devtools_app/lib/src/shared/theme.dart [hunk 4]
```diff
@@ -488,7 +488,7 @@ extension ThemeDataExtension on ThemeData {
       );
 
   TextStyle get devToolsTitleStyle =>
-      textTheme.headline6!.copyWith(color: Colors.white);
+      textTheme.titleLarge!.copyWith(color: Colors.white);
```
Comment (CoderDake) [inline packages/devtools_app/lib/src/shared/theme.dart:491]: same about `titleSmall` here

File: packages/devtools_app/lib/src/framework/scaffold.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/framework/scaffold.dart:455]: Nope, titleLarge.
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/text_theme.dart#L330

File: packages/devtools_app/lib/src/framework/initializer.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/framework/initializer.dart:170]: nope, displaySmall:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/text_theme.dart#L307

**Pull request merged**